plugins/modules/mso_schema_site_external_epg.py

```diff
@@ -160,7 +214,17 @@ def main():
             ops.append(dict(op="remove", path=op_path))
 
     elif state == "present":
-        l3out_dn = "uni/tn-{0}/out-{1}".format(tenant_name, l3out)
+        # Get external EPGs type from template level
+        external_epgs = [e.get("name") for e in schema_obj.get("templates")[template_idx]["externalEpgs"]]
+        if external_epg is not None and external_epg in external_epgs:
```

Comment:
how can external_epg be None during present? Think this part of the if statement is not needed because of the required_if block `["state", "present", ["external_epg"]`